tests/trial_tests/test_fixed_frozen.py

```diff
@@ -0,0 +1,213 @@
+
+    with pytest.raises(ValueError):
+        trial.suggest_loguniform("y", 0.0, 1.0)
+
+
+@parametrize_trial_type
+def test_suggest_discrete_uniform(trial_type: type) -> None:
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 0.9},
+        distributions={"x": FloatDistribution(0.0, 1.0, step=0.1)},
+    )
+    assert trial.suggest_discrete_uniform("x", 0.0, 1.0, 0.1) == 0.9
+
+    with pytest.raises(ValueError):
+        trial.suggest_discrete_uniform("y", 0.0, 1.0, 0.1)
+
+
+@parametrize_trial_type
+def test_suggest_int_log(trial_type: type) -> None:
+
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1},
+        distributions={"x": IntDistribution(1, 10, log=True)},
+    )
+
+    assert trial.suggest_int("x", 1, 10, log=True) == 1
+
+    with pytest.raises(ValueError):
+        trial.suggest_int("x", 1, 10, step=2, log=True)
+
+    with pytest.raises(ValueError):
+        trial.suggest_int("y", 1, 10, log=True)
+
+
+@parametrize_trial_type
+def test_suggest_categorical(trial_type: type) -> None:
+
+    # Integer categories.
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": 1},
+        distributions={"x": CategoricalDistribution((0, 1, 2, 3))},
+    )
+    assert trial.suggest_categorical("x", (0, 1, 2, 3)) == 1
+
+    with pytest.raises(ValueError):
+        trial.suggest_categorical("y", [0, 1, 2, 3])
+
+    # String categories.
+    trial = _create_trial(
+        trial_type=trial_type,
+        params={"x": "baz"},
+        distributions={"x": CategoricalDistribution(("foo", "bar", "baz"))},
+    )
+    assert trial.suggest_categorical("x", ("foo", "bar", "baz")) == "baz"
+
+    # Unknown parameter.
+    with pytest.raises(ValueError):
+        trial.suggest_categorical("y", ["foo", "bar", "baz"])
+
+    # Not in choices.
+    with pytest.raises(ValueError):
+        trial.suggest_categorical("x", ["foo", "bar"])
+
+    # Unknown parameter and bad category type.
+    with pytest.warns(UserWarning):
+        with pytest.raises(ValueError):  # Must come after `pytest.warns` to catch failures.
+            trial.suggest_categorical("x", [{"foo": "bar"}])  # type: ignore
+
+
+@parametrize_trial_type
+@pytest.mark.parametrize(
+    ("suggest_func", "distribution"),
+    [
+        (lambda self, *args: self.suggest_int(*args), IntDistribution(1, 10)),
+        (lambda self, *args: self.suggest_int(*args, log=True), IntDistribution(1, 10, log=True)),
+        (lambda self, *args: self.suggest_int(*args, step=2), IntDistribution(1, 10, step=2)),
+        (lambda self, *args: self.suggest_float(*args), FloatDistribution(1, 10)),
+        (
+            lambda self, *args: self.suggest_float(*args, log=True),
+            FloatDistribution(1, 10, log=True),
+        ),
+        (lambda self, *args: self.suggest_float(*args, step=1), FloatDistribution(1, 10, step=1)),
+    ],
+)
+def test_not_contained_param(
```

Comment:
We can test `Trial` here. It should be the same behavior as `FrozenTrial` and `FixedTrial`.